event_response_tool/layer_loss_duplicator/duplicate_layer_loss.py

```diff
@@ -0,0 +1,348 @@
+            columns={self.event_weights_df.columns[0]: event_id_column}
+        )
+
+        # Perform left-join of weights table and loss set table.
+        # Only events that occur in the weights table will remain.
+        weighted_elt_df = join(
+            self.event_weights_df, elt_df, how="left", on=event_id_column
+        )
+        weighted_elt_df = weighted_elt_df.rename(
+            mapper=lambda name: name.upper(), axis=1
+        )
+        weighted_elt_df.dropna(inplace=True)
+
+        try:
+            if len(weighted_elt_df.columns) < 6:
+                # ELT without secondary uncertainty (maybe AIR)
+                alert.debug(
+                    f"ELT {loss_set.id} without secondary uncertainty"
+                )
+                # Scale mean loss value
+                weighted_elt_df["LOSS"] = (
+                    weighted_elt_df.LOSS * weighted_elt_df.WEIGHT
+                )
+                weighted_elt_df["STDDEVC"] = 0
+                weighted_elt_df["STDDEVI"] = 0
+                # Set exposure value equal to the scaled loss
+                weighted_elt_df["EXPVALUE"] = (
+                    weighted_elt_df.LOSS * weighted_elt_df.WEIGHT
+                )
+                # Remove EventID and Weight columns
+                weighted_elt_df = weighted_elt_df.drop(
+                    ["EVENTID", "WEIGHT"], axis=1
+                )
+                # Set EventId for all entries to 1
+                # NOTE: The platform will automatically combine multiple entries with the same event ID into a single occurrence.
+                weighted_elt_df["EVENTID"] = 1
+            else:
+                # ELT with secondary uncertainty (likely RMS)
+                alert.debug(f"ELT {loss_set.id} with secondary uncertainty")
+                # Scale mean loss value
+                weighted_elt_df["PERSPVALUE"] = (
+                    weighted_elt_df.PERSPVALUE * weighted_elt_df.WEIGHT
+                )
+                # Scale independent standard deviation
+                weighted_elt_df["STDDEVI"] = (
+                    weighted_elt_df.STDDEVI * weighted_elt_df.WEIGHT
+                )
+                # Scale correlated standard deviation
+                weighted_elt_df["STDDEVC"] = (
+                    weighted_elt_df.STDDEVC * weighted_elt_df.WEIGHT
+                )
+                # Scale exposure value
+                weighted_elt_df["EXPVALUE"] = (
+                    weighted_elt_df.EXPVALUE * weighted_elt_df.WEIGHT
+                )
+                # Remove EventID and Weight columns
+                weighted_elt_df = weighted_elt_df.drop(
+                    ["EVENTID", "WEIGHT"], axis=1
+                )
+                # Set EventId for all entries to 1
+                # NOTE: The platform will automatically combine multiple entries with the same event ID into a single occurrence.
+                weighted_elt_df["EVENTID"] = 1
+        except Exception as e:
```

Comment:
This will make it harder to debug - if there are certain errors we expect to see sometimes, can we localize the try/catch to those errors and specific exceptions?

Reply:
I don't expect any specific exceptions since we are only dealing with Pandas dataframes here. I have updated the logger and print statements to reflect the exception now.